script.js

```diff
@@ -1,4 +1,130 @@
+//=====================Rules================================|
+//There will be 2 players and 2 dice.
+//The players will take turns to roll the dice.
+//Both players will then pick the order of each of the two dice results in which they want their 2 digit number to form.
+// After both players have rolled their dice and selected their dice sequence, the player that chooses the higher combined number wins.
+
+// Global variables
+var DICE_ROLL = "DICE_ROLL";
+var DICE_ORDER = "DICE_ORDER";
+var COMPARE_SCORES = "COMPARE_SCORES";
+
+var playerRolls = [];
+var currentPlayer = 1; // Initialize with player 1
+var allPlayerScores = [];
+
+var gameState = DICE_ROLL; // Initialize the game state
+
+var rollDice = function () {
+  console.log("Control flow: start of rollDice()");
+  // Random decimal between 0 and 6
+  var randomDecimal = Math.random() * 6;
+  // Random Integer from 1 to 6
+  var randomInteger = Math.floor(randomDecimal) + 1;
+
+  console.log("rollDice output, randomInteger: ", randomInteger);
+
+  return randomInteger;
+};
+
+var rollDiceForPlayer = function () {
+  console.log("Control flow: start of rollDiceForPlayer()");
+  var counter = 0;
+  while (counter < 2) {
+    playerRolls.push(rollDice());
+    counter = counter + 1;
+  }
+  console.log("rollDiceForPlayer changes, playerRolls: ", playerRolls);
+
+  return (
+    "Welcome to the Beat That! Game✨<br><br>You have rolled:<br>🎲Dice One: " +
+    playerRolls[0] +
+    "  🎲Dice Two: " +
+    playerRolls[1] +
+    ".<br><br>Now, please input either '1' or '2' to select the corresponding dice to be used as the first digit of your final value."
+  );
+};
+
+// Input validation
+var getPlayerScore = function (input) {
+  if (input !== "1" && input !== "2") {
+    return (
+      "❌Error Alert❌<br><br> Please only input '1' or '2' to choose the dice to use as first digit.<br><br>Your dice rolls are:<br>🎲Dice One: " +
+      playerRolls[0] +
+      "  🎲Dice Two: " +
+      playerRolls[1] +
+      "."
+    );
+  }
+
+  var playerScore;
+  if (input == 1) {
+    playerScore = Number(playerRolls[0] + "" + playerRolls[1]);
+  } else if (input == 2) {
+    playerScore = Number(playerRolls[1] + "" + playerRolls[0]);
+  }
```

Comment:
Interesting way to handle this logic. I would propose another solution though, since we already know `playerRolls` are all numbers, maybe we can handle this situation using just numbers:
```
playerScore = playerRolls[0] * 10 + playerRolls[1]
```
Both works! just my own preference to not have to manually cast the types around